### PR TITLE
Bugfix/nex 417/compilation report are not getting displayed

### PR DESCRIPTION
**Pull request by k-v-vladimir**
https://oat-sa.atlassian.net/browse/NEX-417
[BE] Compilation report are not getting displayed

The detailed report is returned to front-end

### CODE REVIEW

**Comment by boajer** [comment]
Please not forget about version bump for extension

File: model/publication/LookupSnapshot.php [hunk 1]
```diff
@@ -19,7 +19,7 @@
 
 namespace oat\taoDeliverConnect\model\publication;
 
-use common_report_Report;
+use \common_report_Report as Report;
```
Comment (mike-ionut-mihai-sandu-tao) [inline model/publication/LookupSnapshot.php]: nice 👍

File: model/publication/LookupSnapshot.php [hunk 2]
```diff
@@ -29,6 +29,8 @@ class LookupSnapshot
         self::STATUS_SUCCESS,
         self::STATUS_FAILED,
     ];
+    const REPORT_TYPE = 'type';
```
Comment (mike-ionut-mihai-sandu-tao) [inline model/publication/LookupSnapshot.php]: not mandatory but consider to use the visibility for constants also, i assume that you are using these constants only inside here, so private would be ok

File: model/publication/LookupSnapshot.php [hunk 3]
```diff
@@ -67,12 +69,22 @@ public function getStatus()
     }
 
     /**
+     * @param bool $generateReport Returns \common_report_Report[] if true
      * @return array|common_report_Report[]
-     * @todo add param to convert into cur-gen report collection
      */
-    public function getReports()
+    public function getReports($generateReport = false)
```
Comment (mike-ionut-mihai-sandu-tao) [inline model/publication/LookupSnapshot.php]: are you sure you want to change the signature of the function? 
what do you think about creating another function called `generateReport`, that will be called if the getter returns null

File: model/publication/LookupSnapshot.php [hunk 4]
```diff
@@ -67,12 +69,22 @@ public function getStatus()
...
     {
-        return $this->reports;
+        if (!$generateReport) {
+            return $this->reports;
+        }
+
+        $reports = [];
+
+        foreach ($this->reports as $report) {
```
Comment (mike-ionut-mihai-sandu-tao) [inline model/publication/LookupSnapshot.php]: i think this should be inside the `generateReport`, or `buildReport`, inside a `ReportBuilder` class

**Comment by boajer** [comment]
@k-v-vladimir please resolve conflict

**Pull request merged**